src/elastio-snap.c

```diff
@@ -1499,6 +1537,7 @@ static void task_work_flush(void){
 /******************************FILE OPERATIONS*******************************/
 
 static inline void file_close(struct file *f){
+	file_unlock_mark_dirty(f);
 	filp_close(f, NULL);
```

Comment:
Why do we need to put the inode on the super block's dirty list? Will it cause an fsck error then on mounts of a snapshot? I hope no )

Reply:
This handles the situation when we umount the device and the OS has just [randomly] synced our immutable flag and updated the attributes. This will cause a situation when the COW file on the unmounted disk is immutable, which is surely something we don't want to happen. Hence, we mark it as dirty to be updated when a such event as umount happens: this will reset the immutable flag and the attributes will be updated correctly.

Reply:
About the question about copying the COW file in the active mode: it will give you a 'Permission denied' error. Not sure if we need such a test, probably worth discussing this.